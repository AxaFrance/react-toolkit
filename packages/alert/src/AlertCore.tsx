import React, { MouseEventHandler, ReactNode } from 'react';
import {
  withClassDefault,
  withClassModifier,
  WithClassModifierOptions,
  compose,
  identity,
} from '@axa-fr/react-toolkit-core';

const DEFAULT_CLASSNAME = 'af-alert';

export type AlertCoreComponentProps = {
  title: string;
  iconClassName?: string;
  onClose?: MouseEventHandler<HTMLButtonElement>;
  className?: string;
  children?: ReactNode;
} & WithClassModifierOptions;
export type AlertCoreProps = AlertCoreComponentProps;

const AlertCoreRaw = ({
  className,
  onClose,
  iconClassName,
  title,
  children,
}: AlertCoreComponentProps) => (
  <div className={className} role="alert">
    <div className="af-alert__title">
      <div className="af-alert__title-icon">
        <i className={iconClassName} />
      </div>
      <div className="af-alert__title-text">{title}</div>
      {onClose && (
        <button
          type="button"
          className="af-alert__title-icon-close"
          onClick={onClose}>
          <i className="glyphicon glyphicon-close" />
        </button>
      )}
    </div>
    {children && (
      <div className="af-alert__content">
        <div className="af-alert__content__left" />
        <div className="af-alert__content__right">{children}</div>
      </div>
    )}
  </div>
);

const enhance = compose(
  identity<AlertCoreComponentProps>(),
  withClassDefault(DEFAULT_CLASSNAME),
  withClassModifier()
);

const Enhanced = enhance(AlertCoreRaw);
Enhanced.displayName = 'AlertCore';
export default Enhanced;
