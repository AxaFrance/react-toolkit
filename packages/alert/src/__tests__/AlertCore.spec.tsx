import React from 'react';
import { render } from '@testing-library/react';
import AlertCore from '../AlertCore';

describe('<AlertCore />', () => {
  it('renders an error alert with exclamation icon and title', () => {
    const { asFragment } = render(
      <AlertCore
        iconClassName="glyphicon glyphicon-exclamation-sign"
        classModifier="error"
        title="Les caves et les garages situés dans le même corps de bâtiment que le logement assuré sont garantis"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Shoud contain close element when onClose props exists', () => {
    const onCloseMock = jest.fn();
    const { getByRole } = render(
      <AlertCore title="Succès" iconClassName="success" onClose={onCloseMock} />
    );
    expect(getByRole('button')).toBeInTheDocument();
  });
});
