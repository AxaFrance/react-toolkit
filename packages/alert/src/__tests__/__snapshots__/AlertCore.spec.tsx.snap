// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertCore /> renders an error alert with exclamation icon and title 1`] = `
<DocumentFragment>
  <div
    class="af-alert af-alert--error"
    role="alert"
  >
    <div
      class="af-alert__title"
    >
      <div
        class="af-alert__title-icon"
      >
        <i
          class="glyphicon glyphicon-exclamation-sign"
        />
      </div>
      <div
        class="af-alert__title-text"
      >
        Les caves et les garages situés dans le même corps de bâtiment que le logement assuré sont garantis
      </div>
    </div>
  </div>
</DocumentFragment>
`;
